internal/logging/file.go

```diff
@@ -0,0 +1,210 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ActiveState/cli/internal/condition"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/rollbar/rollbar-go"
+)
+
+var defaultMaxEntries = 1000
+
+type entry struct {
+	ctx     *MessageContext
+	message string
+	args    []interface{}
+}
+
+type fileHandler struct {
+	formatter Formatter
+	file      *os.File
+	mu        sync.Mutex
+	verbose   safeBool
+	wg        *sync.WaitGroup
+	queue     chan entry
+	quit      chan struct{}
+}
+
+func newFileHandler() *fileHandler {
+	handler := fileHandler{
+		DefaultFormatter,
+		nil,
+		sync.Mutex{},
+		safeBool{},
+		&sync.WaitGroup{},
+		make(chan entry, defaultMaxEntries),
+		make(chan struct{}),
+	}
+	go handler.start()
+	return &handler
+}
+
+func (l *fileHandler) start() {
+	for {
+		select {
+		case entry := <-l.queue:
+			l.emit(entry.ctx, entry.message, entry.args)
+			l.wg.Done()
+		case <-l.quit:
+			return
```

Comment:
```suggestion
			close(l.queue)
			for range l.queue {
			}
			return
```

Reply:
I'm sending mixed messages here, sorry. If you want to send all the messages before collapsing, the for loop needs to call `l.emit`. The above will just drain cleanly and then collapse.

Reply:
Yeah, I was missing this part in my original changeset. I've added an `l.emit` call to the for loop. Thanks for the suggestion!